src/ng/compile.js

```diff
@@ -969,6 +969,17 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     return require;
   }
 
+  function getDirectiveRestrict(directive, name) {
+    if (directive.restrict != null && !(isString(directive.restrict) && /[EACM]/.test(directive.restrict))) {
```

Comment:
I don't think so, the various cases (that I've thought of) would be...
* null/undefined => same as before
* non-string => previously it would crash & log every time `addDirective` loops over that directive, and the directive would not get used.  Now it logs once here and avoids adding it to the directives list.
* string containing none of E, A, C, M => previously it would never match anything. Now it logs once here and avoids adding it to the directives list.

Using `/^[EACM]+$/` for the regex would make this a breaking change (and would fully validate the string, unlike the current solution that allows things like `restrict: "E----#(*&#$@(  "`).

Things I didn't think about until right now...
* previously `restrict: ['E', 'A']` was equivalent to `restrict: 'EA'` because `addDirective` just blindly calls `directive.restrict.indexOf(location)`